data_validation/partition_builder.py

```diff
@@ -311,7 +333,6 @@ def _add_partition_filters(
         Returns:
             yaml_configs_list (List[Dict]): List of YAML configs for all tables
         """
-
```

Comment:
I think we need more clarity on what these input data structures are. Not sure I understand what this means...

List for each ConfigManager, containing a List of partition filters which each has a List for source and target filters?
[ConfigManager1 [ PartitionFilter1 [source: filter], [target:filter], PartitionFilter2 [source: filter], [target:filter] ] ]

Would it be cleaner to create a class called PartitionFilter that stores the source and target filter? So we can easily access them (PartitionFilter.source_filter) without nested lists?

As for the return value - how is the Dict structured?
List of Dict {"yaml_files": {{"target_file_name": "file.yaml", "yaml_config": yaml_config}}